File: G_Projects/G_Inventory_Management_System/ProductFactory/ConcreteProducts/ClothingProduct.java
package G_Projects.G_Inventory_Management_System.ProductFactory.ConcreteProducts;

import G_Projects.G_Inventory_Management_System.CommonEnums.ProductEnum;
import G_Projects.G_Inventory_Management_System.ProductFactory.Product;

public class ClothingProduct extends Product{
    private String size;
    private String color;

    public ClothingProduct(String id, String name, double price, int quantity, int threshold){
        setId(id);
        setName(name);
        setPrice(price);
        setQuantity(quantity);
        setThreshold(threshold);
        setCategory(ProductEnum.CLOTHING);
    }

    private String getSize(){
        return size;
    }
    private void setSize(String size){
        this.size = size;
    }

    private String getColor(){
        return color;
    }
    private void setColor(String color){
        this.color = color;
    }
}

File: Python/A_Project/D_Inventory_Management_System/ProductFactory/ConcreteProducts/clothing_product.py
from CommonEnums.product_enum import ProductEnum
from ProductFactory.product import Product


class ClothingProduct(Product):
    def __init__(self, product_id, name, price, quantity, threshold):
        super().__init__()
        self._size = None
        self._color = None
        self.set_id(product_id)
        self.set_name(name)
        self.set_price(price)
        self.set_quantity(quantity)
        self.set_threshold(threshold)
        self.set_category(ProductEnum.CLOTHING)

    def get_size(self):
        return self._size

    def set_size(self, size: str):
        self._size = size

    def get_color(self):
        return self._color

    def set_color(self, color: str):
        self._color = color
